### DISCUSSION TITLE
Is there any clean way to append items to a one-to-many relationship ?

### DISCUSSION THREAD

**Discussion by chipseater** [supabase/supabase#7032] [Questions] [answered]
Hi,
Very innocent question: if we have an array of items in a table cell, is there any way to append data to it without having the need to fetch the data of the whole row to get its content, in order to finally push the updated data to the cell ? I have a column named content, which is an array of file identifiers, and I want to append some new data to it. Is there any clean way to do this action or we need to perform a `select()` before an `update()` ?

**Comment by GaryAustin1** [comment, marked as answer]
You would need to use an rpc function to do the update in SQL as there are several useful array functions in Postgres.
https://www.postgresql.org/docs/current/functions-array.html

**Reply by chipseater** [reply]
Thank you for the response, I will setup that soon !

**Comment by dshukertjr** [comment]
As GaryAustin1 said, you would need to create an Postgres function and call it using the `rpc()` method. 

You can see a detailed examples here:
https://github.com/supabase/supabase/discussions/2771
